File: Source/Entities/QuarterRotateBooster.cs
using System;
using Celeste.Mod.Entities;
using Microsoft.Xna.Framework;
using Monocle;

namespace Celeste.Mod.CaeruleaHelper.Entities;

[Tracked]
[CustomEntity("CaeruleaHelper/QuarterRotateBooster")]
public class QuarterRotateBooster : Booster
{
    private readonly Sprite TwistSprite;
    private readonly string directory;
    private readonly float radians;

    public QuarterRotateBooster(EntityData data, Vector2 offset) : base(data.Position + offset, data.Bool("red"))
    {
        directory = data.Attr("twistSprite", "objects/caerulea/revbooster/twist/twist").TrimEnd(['/']);
        Add(TwistSprite = new Sprite(GFX.Game, directory));
        TwistSprite.AddLoop("twist", "", .1f);
        TwistSprite.Play("twist");
        TwistSprite.CenterOrigin();
        radians = Calc.DegToRad * data.Float("degrees", 90);
    }
    private void Use(Player player)
    {
        player.Speed = player.Speed.Rotate(radians);
        player.DashDir = player.DashDir.Rotate(radians);
    }
    public static void HookUse(On.Celeste.Booster.orig_PlayerBoosted orig, Booster self, Player player, Vector2 direction)
    {
        orig(self, player, direction);
        if (self is QuarterRotateBooster qr) qr.Use(player);
    }
    public static void Load()
    {
        On.Celeste.Booster.PlayerBoosted += HookUse;
    }
    public static void Unload()
    {
        On.Celeste.Booster.PlayerBoosted -= HookUse;
    }
}

File: Loenn/entities/quarter_rotate_booster.lua
local quarterRotateBooster = {}

quarterRotateBooster.name = "CaeruleaHelper/QuarterRotateBooster"
quarterRotateBooster.depth = -100
quarterRotateBooster.placements = {
    {
        name = "quarter_rotate_booster",
        data = {
            red = false,
            degrees = 90,
            twistSprite = "objects/caerulea/revbooster/twist/twist"
        }
    }
}

function quarterRotateBooster.texture(room, entity)
    local red = entity.red

    if red then
        return "objects/booster/boosterRed00"

    else
        return "objects/booster/booster00"
    end
end

return quarterRotateBooster
